### PR TITLE
crypto/tls: Add ECHOffered bool to ConnectionState.

### PR DESCRIPTION
**Pull request by chris-wood**
See #104.

### CODE REVIEW

File: src/crypto/tls/common.go [hunk 1]
```diff
@@ -338,6 +338,9 @@ type ConnectionState struct {
 	// accepted by the server.
 	ECHAccepted bool
 
+	// ECHOffered is set if the ECH extension was offered by the client.
```
Comment (cjpatton) [inline src/crypto/tls/common.go]: To match the verbage elsewhere in the code, this should say
```suggestion
	// ECHOffered is set if the ECH extension is present in the ClientHello.
	// This means the client has offered ECH or sent GREASE ECH.
```
Also, we might consider giving it a slightly less ambiguous name, like `ECHExtensionPresent` or something. Alternatively, you could add two flags, one for ECH offered and another for ECH greased.

File: src/crypto/tls/conn.go [hunk 2]
```diff
@@ -1475,6 +1475,7 @@ func (c *Conn) connectionStateLocked() ConnectionState {
 	state.SignedCertificateTimestamps = c.scts
 	state.OCSPResponse = c.ocspResponse
 	state.ECHAccepted = c.ech.accepted
+	state.ECHOffered = c.ech.offered
```
Comment (cjpatton) [inline src/crypto/tls/conn.go]: `c.ech.offered == true` if ECH was offered by the client and rejected or accepted by the server.
`c.ech.greased == true` if the client sent GREASE ECH.

Hence, the condition you're after is `c.ech.offered || c.ech.greased`.

**Review by cjpatton — CHANGES_REQUESTED** [review]

File: src/crypto/tls/conn.go (hunk 2 above)
Comment (chris-wood) [reply src/crypto/tls/conn.go]: Fixed.

**Review by cjpatton — APPROVED** [review]
LGTM. Be sure to squash. Also, our convention for commit messages is to not include the period "."

**Comment by chris-wood** [comment]
@cjpatton squashed and ready to go. @claucece, can you please take a look?

**Review by claucece — APPROVED** [review]
LGTM ;)

**Pull request merged**